apple/internal/partials/debug_symbols.bzl

```diff
@@ -201,26 +210,27 @@ def _bundle_dsym_files(
     outputs = []
 
     if debug_outputs_provider:
+        dsym_binaries = {}
         for (arch, arch_outputs) in debug_outputs_provider.outputs_map.items():
-            outputs.append(_copy_dsym_into_declared_bundle(
-                actions = actions,
-                arch = arch,
-                debug_output_filename = debug_output_filename,
-                dsym_binary = arch_outputs["dsym_binary"],
-                dsym_bundle_name = dsym_bundle_name,
-                executable_name = executable_name,
-            ))
+            dsym_binaries[arch] = arch_outputs["dsym_binary"]
```

Comment:
@thii Since this is supposed to work for xcframeworks, what if it gets an arm64 simulator and an arm64 device? lipo won't work in that case.

Reply:
Ahh, I think this is called for each platform, so we should be good. I'll run some tests.